src/query/processor.rs

```diff
@@ -108,7 +146,7 @@ impl<T: Transport + Clone> Processor<T> {
     /// * query is not registered yet
     /// * creates gateway and network
     /// * registers query
-    pub async fn prepare(&self, req: &PrepareQuery) -> Result<(), PrepareQueryError> {
+    pub async fn prepare(&self, req: PrepareQuery) -> Result<(), PrepareQueryError> {
```

Comment:
what's the reasoning behind having a different error type for each command?

Reply:
no actual reason, I added commands one by one. for now it works, once I have the full implementation of query processor, some errors types can be merged